app/assets/javascripts/sitewide/file_preview.js

```diff
@@ -288,17 +317,21 @@ var FilePreviewModal = (function() {
     $('.file-save-link').off().click(function(ev) {
       var imageBlob;
       var imageDataURL;
-      var imageFormat;
+      var imageParams;
       var dataUpload = new FormData();
       var blobArray;
       var bytePosition;
 
       ev.preventDefault();
       ev.stopPropagation();
 
-      imageFormat = (data['mime-type'] === 'image/png') ? 'png' : 'jpeg';
+      if (data['mime-type'] === 'image/png') {
+        imageParams = { format: 'png' };
+      } else {
+        imageParams = { format: 'jpeg', quality: (data.quality / 100) };
```

Comment:
What if imagemagick returns quality value in different format, lets say in percents?

Reply:
For jpeg it use quality format from 1 to 100.

Reply:
And if in new version it moves to 0.00 - 0.99 format?

Reply:
Add check for format before saving